.github/workflows/docker_publish.yml

```diff
@@ -0,0 +1,49 @@
+
+on:
+  push:
+    branches: ['gh_action_gh_container_registry']  # todo replace with main once we know this is working
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+
+    env:
```

Comment:
non-blocking suggestion: Could we make this an `env` of the step in question? https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsenv

This would localize it to that place in the file and make its purpose more obvious

Reply:
I've had a go at moving this and the actions appear to be running as before, thanks for the link that helped